core/pkg/server/stream_init.go

```diff
@@ -181,7 +181,11 @@ func NewFileStream(
 		Logger:            logger,
 		Printer:           printer,
 		ApiClient:         fileStreamRetryClient,
-		TransmitRateLimit: rate.NewLimiter(rate.Every(15*time.Second), 1),
+		TransmitRateLimit: rate.NewLimiter(rate.Every(filestream.DefaultTransmitInterval), 1),
```

Comment:
i would suggest moving this logic into the NewFileStream
(if the `TransmitRateLimit` is nil use default otherwise use provided argument)